src/Conversion/ONNXToKrnl/Tensor/PrintSignature.cpp

```diff
@@ -50,15 +50,22 @@ struct ONNXPrintSignatureLowering
           op, msg + "(no tensors)\n%e", noneVal);
       return success();
     }
+    // Control how the tensor will be printed
+    // Print the only the shape.
+    std::string printControl = "%e";
+    if (printSignatureOp.getPrintData() == 1) {
+      // The data of tensor will be printed
+      printControl = "%d";
+    }
     Value lastVal = printVal.pop_back_val();
     // Print all but the last one.
     for (Value oper : printVal) {
-      create.krnl.printTensor(msg + ", %t%e", oper);
+      create.krnl.printTensor(msg + ", %t" + printControl, oper);
```

Comment:
I believe the "%e" is still recommended for the end of the string. This is because we sometimes have issues with the `\0` end of string (never successfully debugged), so we recommend to add an explicit end of string.

Reply:
Fixed